### ISSUE TITLE
MetadataReader.GetNamespaceDefinition is crashing on framework assemblies

### ISSUE THREAD

**Issue by jkotas** [dotnet/corefx#2244]
Repro:
```
var peReader = new PEReader(File.OpenRead(@"Microsoft.CSharp.dll"));
var reader = peReader.GetMetadataReader();
var root = reader.GetNamespaceDefinitionRoot();
```

Result:
```
System.ArgumentException: An item with the same key has already been added. at
System.Collections.Generic.Dictionary<System.Reflection.Metadata.NamespaceDefinitionHandle, System.__Canon>.Add(System.Reflection.Metadata.NamespaceDefinitionHandle key, System.__Canon value)
System.Reflection.Metadata.Ecma335.NamespaceCache.PopulateNamespaceTable()
```

This bug makes the namespace feature of metadata reader useless.

**Comment by jkotas** [comment]
The crash is caused by same suffix shared by multiple namespaces, like:

```
namespace FxResources.Microsoft.CSharp
{
    public class Resources
    { }
}

namespace Microsoft.CSharp
{
    public class Program
    { }
}
```

I have shared the test update to include this case at https://github.com/jkotas/corefx/commit/7ed83663ef0662a29f21b5d6ea4e6c35b14d45a6

**Comment by jkotas** [comment]
@tmat @nguerrera Thoughts on how to fix it?

**Comment by nguerrera** [comment]
I'll take a look.

**Comment by tmat** [comment]
Oops. We should really get Roslyn to use the reader's namespace API instead of its own namespace tree builder. We'd likely catch this earlier.

**Comment by nguerrera** [comment]
Yes. I have a potential fix #2245.
